File: baekjoon/python/1935.py
import sys; input =sys.stdin.readline
N = int(input())

postfix = input().rstrip()
postfix_dict = {} # 알파벳을 키 값으로 두고 숫자를 데이터로 초기화

for i in range(N):
    num = int(input())
    postfix_dict[chr(65 + i)] = num

calc = [] # 후위표기식을 계산할 스택
for c in postfix:
    if c in postfix_dict: calc.append(int((postfix_dict[c])))
    else:
        if c == "*": calc.append(calc.pop() * calc.pop())
        if c == "/": 
            back = calc.pop()
            calc.append(calc.pop() / back)
        if c == "+": calc.append(calc.pop() + calc.pop())
        if c == "-": 
            back = calc.pop()
            calc.append(calc.pop() - back)

print("%.2f" % calc[0])

File: baekjoon/javascript/1935.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", line => {
  input.push(line);
}).on("close", () => {
  const N = +input[0];
  const postfix = input[1];
  const postfixMap = new Map();

  for (let i = 2; i < input.length; i++) {
    const num = input[i];
    postfixMap.set(String.fromCharCode(63 + i), num);
  }

  const calc = [];
  for (let c of postfix) {
    if (postfixMap.has(c)) calc.push(+postfixMap.get(c));
    else
      switch (c) {
        case "*": {
          calc.push(calc.pop() * calc.pop());
          break;
        }
        case "/": {
          const back = calc.pop();
          calc.push(calc.pop() / back);
          break;
        }
        case "+": {
          calc.push(calc.pop() + calc.pop());
          break;
        }
        case "-": {
          const back = calc.pop();
          calc.push(calc.pop() - back);
          break;
        }
        default:
          break;
      }
  }
  console.log(calc[0].toFixed(2));
});
